common/util.go

```diff
@@ -519,10 +519,12 @@ func CreateHistoryStartWorkflowRequest(
 	domainID string,
 	startRequest *types.StartWorkflowExecutionRequest,
 	now time.Time,
+	partitionConfig map[string]string,
```

Comment:
I'm worried about changes such as this will presumably need to include subsequent changes once the rest of the feature is added in which will lead to a lot of conflicts as both commits will be editing the same line. Are you sure you want to do it this way? 

What about an alternative where we merge: 
1. Add to, or expand existing Types in a non-breaking way
2. add the Libraries and other dependencies we'll need
3. Doing all these 'breaking' changes in a single commit where the CreateHistoryStartWorkflowRequest immediately uses the data it's expecting, rather than requirign a subsequent refactor

Reply:
There are no conflicts from the feature branch.